homeassistant/components/backup_sftp/config_flow.py

```diff
@@ -0,0 +1,175 @@
+"""Config flow to configure the SFTP Backup Storage integration."""
+
+from __future__ import annotations
+
+from pathlib import Path
+from typing import Any
+
+from asyncssh import KeyImportError, SSHClientConnectionOptions, connect
+from asyncssh.misc import PermissionDenied
+from asyncssh.sftp import SFTPNoSuchFile, SFTPPermissionDenied
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+from homeassistant.exceptions import ConfigEntryError
+from homeassistant.helpers.selector import (
+    TextSelector,
+    TextSelectorConfig,
+    TextSelectorType,
+)
+
+from . import SFTPConfigEntryData
+from .const import (
+    CONF_BACKUP_LOCATION,
+    CONF_HOST,
+    CONF_PASSWORD,
+    CONF_PORT,
+    CONF_PRIVATE_KEY_FILE,
+    CONF_USERNAME,
+    DOMAIN,
+    LOGGER,
+)
+
+
+class SFTPFlowHandler(ConfigFlow, domain=DOMAIN):
+    """Handle an SFTP Backup Storage config flow."""
+
+    def _check_pkey_and_password(self, user_input: dict[str, Any]) -> dict[str, Any]:
+        """Check if user provided either one of password or private key.
+
+        Additionally, check if private key exists and make sure it starts
+        with `/config` if full path is not provided by user.
+
+        Returns: user_input object with edited private key location, if edited.
+        """
+        # If both password AND private key are not provided, error out.
+        # We need at least one to perform authentication.
+        if (
+            bool(user_input.get(CONF_PASSWORD)) is False
+            and bool(user_input.get(CONF_PRIVATE_KEY_FILE)) is False
+        ):
+            raise ConfigEntryError(
+                "Please configure password or private key file location for SFTP Backup Storage."
+            )
+
+        if bool(user_input.get(CONF_PRIVATE_KEY_FILE)):
+            # If full path to private key is not provided,
+            # fallback to /config as a main directory to look for pkey.
+            if not user_input[CONF_PRIVATE_KEY_FILE].startswith("/"):
+                user_input[CONF_PRIVATE_KEY_FILE] = (
+                    "/config/" + user_input[CONF_PRIVATE_KEY_FILE]
+                )
```

Comment:
We should access the files securely. @farmio has something built in KNX where the user can upload their key, but I think we can be smarter with this